pkg/controllers/common/deployment.go

```diff
@@ -2,13 +2,51 @@ package common
 
 import (
 	"fmt"
+	"strings"
 
 	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	corelistersv1 "k8s.io/client-go/listers/core/v1"
 
 	operatorv1 "github.com/openshift/api/operator/v1"
 )
 
-func CheckDeploymentReady(deployment *appsv1.Deployment, conditionPrefix string) []operatorv1.OperatorCondition {
+// DeploymentPodsStatus return detailed information about deployment pods and the containers
+// TODO: Move this to library-go
+func DeploymentPodsStatus(deployment *appsv1.Deployment, podClient corelistersv1.PodLister) ([]string, error) {
+	deploymentPods, err := podClient.Pods(deployment.Namespace).List(labels.SelectorFromSet(deployment.Spec.Template.Labels))
+	if err != nil {
+		return nil, err
+	}
+	deploymentPodsStates := []string{}
+	for i := range deploymentPods {
+		containerStates := []string{}
+		for _, c := range append(deploymentPods[i].Status.ContainerStatuses, deploymentPods[i].Status.InitContainerStatuses...) {
+			switch {
+			case c.State.Running != nil:
+				containerStates = append(containerStates, fmt.Sprintf("container %q is running since %s", c.Name, c.State.Running.StartedAt))
```

Comment:
do we need the running ones?

Reply:
yes, if 2 and running I want to know when they started and why there is 1 pending for 20 minutes :-) more details here the better.